src/connector/src/parser/mod.rs

```diff
@@ -616,6 +640,7 @@ const MAX_ROWS_FOR_TRANSACTION: usize = 4096;
 async fn into_chunk_stream<P: ByteStreamSourceParser>(mut parser: P, data_stream: BoxSourceStream) {
     let columns = parser.columns().to_vec();
 
+    let mut heartbeat_builder = SourceStreamChunkBuilder::with_capacity(columns.clone(), 0);
```

Comment:
I would suggest making it new type around `Builder` like `HeartbeatBuilder` and only expose `xxx_heartbeat` methods. Having `xxx_heartbeat` on a normal `ChunkBuilder` looks kind of weird to me.